### PR TITLE
add "Review" button to TEC (Admin)

### PR DESCRIPTION
**Pull request by kevinmram**
add "Review" button to TEC (Admin), and view user submission w/ Set to Approved/Rejected.

- However, the attached image doesn't display and is replaced with a default image...

https://github.com/user-attachments/assets/de4a618b-e591-455e-a268-b79645b0cb75

### CODE REVIEW

**Comment by andrew032011** [comment]
> - However, the attached image doesn't display and is replaced with a default image...

https://github.com/cornell-dti/idol/pull/655

File: frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx [hunk 1]
```diff
@@ -120,39 +83,47 @@ const TeamEventCreditReview = (props: {
...
-        >
-          Reject
-        </Button>
         <Input
           type="text"
           placeholder="Reason for reject"
+          value={reason}
           onChange={(e) => setReason(e.target.value)}
         />
+        {/* Show 'Set to Rejected' only if the current status is approved */}
```
Comment (patriciaahuang) [inline frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx:101]: Remove comment

File: frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx [hunk 2]
```diff
@@ -120,39 +83,47 @@ const TeamEventCreditReview = (props: {
...
+        {currentStatus === 'approved' && (
+          <Button basic color="red" disabled={reason === ''} onClick={rejectCreditRequest}>
+            Set to Rejected
+          </Button>
+        )}
+        {/* Show 'Set to Approved' only if the current status is rejected */}
```
Comment (patriciaahuang) [inline frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx:107]: Remove comment

**Review by patriciaahuang — CHANGES_REQUESTED** [review]
Good job on this! I think reviewing a rejected team event submission shouldn't show the textbox for rejection reason, since the reviewer wouldn't have a rejection reason when approving the submission (and it'll be set to null no matter what reason they put).

Screenshot 2024-10-27 at 4 19 58 PM

File: frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx [hunk 3]
```diff
@@ -119,40 +78,47 @@ const TeamEventCreditReview = (props: {
...
+        {currentStatus === 'pending' && (
+          <>
+            <Button basic color="green" onClick={approveCreditRequest}>
+              Approve
+            </Button>
+            <Button basic color="red" disabled={reason === ''} onClick={rejectCreditRequest}>
+              Reject
+            </Button>
+          </>
+        )}
```
Comment (patriciaahuang) [inline frontend/src/components/Admin/TeamEvent/TeamEventCreditReview.tsx:118]:
```suggestion
        {currentStatus !== 'rejected' && (
          <Button basic color="red" disabled={reason === ''} onClick={rejectCreditRequest}
          >
            {currentStatus === 'pending' ? 'Reject' : 'Set to Rejected'}
          </Button>
        )}
        {currentStatus !== 'approved' && (
          <Button basic color="green" onClick={approveCreditRequest}
          >
            {currentStatus === 'pending' ? 'Approve' : 'Set to Approved'}
          </Button>
        )}
```

**Review by patriciaahuang — APPROVED** [review]
Left a suggestion to make the logic more concise, but otherwise looks good!

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**